### ISSUE TITLE
Update releases JSON format

### ISSUE THREAD

**Issue by marcwrobel** [endoflife-date/release-data#51]
Currently the JSON format for releases in `/releases` is :

```json
{
  "<version_1>: "<date_1>",
  "<version_2>: "<date_2>",
...
  "<version_n>: "<date_n>"
}
```

This format is not extensible : only a date can be provided. Using a format like the following one could make it extensible (e.g. open to new properties that could eventually used for more automation):

```json
{
  "versions": [
    {
      "number": "<version_1>"
      "date": "<date_1>"
    },
    {
      "number": "<version_2>"
      "date": "<date_2>"
    },
...
    {
      "number": "<version_n>"
      "date": "<date_n>"
    }
  ]
}
```

This will make it possible to add more properties at product level and at version level.

I do not particularly have "extensions" ideas for the time being, but I have the feeling we should not wait to do that. The more we will add automation scripts, the more work it will be if one day we want to change the format.

**Comment by marcwrobel** [comment]
This issue is linked to https://github.com/endoflife-date/releases.json, but that may be a bit difficult to implement such formats.

**Comment by marcwrobel** [comment]
Having versions grouped under their cycle would be best if possible. I would make possible for instance to automate the retrieval of EOL date (for example for https://learn.microsoft.com/en-us/lifecycle/products/?terms=Azure%20DevOps).

**Comment by captn3m0** [comment]
Been wanting to do this (See [Step 1](https://github.com/endoflife-date/endoflife.date/issues/1898#issuecomment-1327038630) on this discussion for eg), so big +1 for creating this.

I like the proposed schema, can be improved with additional fields (such as link for eg) over time.

One important extension is to support EOL and development dates. For example, for several manual scrapes, like:

- COS
- GKE
- Palo Alto Pages
- EKS
- HAProxy
- ROS

we scrape pages that already include the EOL data. This is data that can automatically be patched (if upstream changes) using latest.py.

-1 for release cycle grouping, since it can get quite confusing. We have releaseCycle labels that latest.py uses, and it still has to make some guesses.

> I do not particularly have "extensions" ideas for the time being, but I have the feeling we should not wait to do that. The more we will add automation scripts, the more work it will be if one day we want to change the format.

+1


(Puts long-term hat on). I think of this as an ingredient dataset to our main repo. Ideally, this repo by itself will be helpful to other projects by itself as well. So ideally, this would closely match the releases.json format by itself, with any additional bits getting patched on top by the main repo. But a core-idea is for this to be a repo containing all versions, which we aren't accounting for in our releases.json schema (yet). So maybe something to patch there as well? We already have a `supportedReleases` key, having another `releases` key (same as `versions` in your proposal above) won't make much difference - it's gonna be optional anyway.


Since this would be a breaking change for `latest.py`, maybe we can patch `latest.py` first to support both old and new schema. That lets us incrementally change the automation here over a few separate PRs.

**Comment by marcwrobel** [comment]
We could also automatically create/update purls using repology (https://github.com/endoflife-date/endoflife.date/pull/1778#issuecomment-1359140137).

**Comment by captn3m0** [comment]
Yes, that is definitely needed. Need to write the script for that. Unsure about keeping the generated PURLs separate, or overwriting the md files (like how we do with latest.py). Probably going with latter.

**Comment by marcwrobel** [comment]
Yes, overwriting is better IMHO.

**Comment by marcwrobel** [comment]
Another improvement: we could also indicate the origin of the data (npmjs, maven, git...).

**Comment by captn3m0** [comment]
We can provide a list of PURLs for the data-source. `pkg:rubygems/bundler` points to https://rubygems.org/gems/bundler, but is more standardized.

Another use-case for our frontend would be to directly link to relevant releases from the latest column.

**Comment by captn3m0** [comment]
use-case for DotNet to track EOL: https://github.com/endoflife-date/endoflife.date/issues/2234

**Comment by marcwrobel** [comment]
This should have been closed with #274. Release-level data will be supported after #297 is merger.

Other ideas, such as supporting product-level data (for PURLs) or data origin will be handled in separate issues if there is some need.
